### PR TITLE
VReplication: Support reversing read-only traffic in vtctldclient

### PR DESCRIPTION
**Pull request by mattlord**
## Description

This PR fixes a bug in `vtctldclient`, in that it did not support a needed feature that the legacy `vtctlclient` did (see the [client transition docs](https://vitess.io/docs/reference/vtctldclient-transition/)). Specifically, support for using `ReverseTraffic` when traffic for read-only tablet types (REPLICA, RDONLY) have been switched BUT writes (PRIMARY tablet type) have not _ever been_ before.

h/t to @rohit-nayak-ps for the additional tests that were a huge help! 🙇 

Using the manual test from below, this fails with `vtctldclient` but works with `vtctlclient`:
```
❯ vtctldclient MoveTables --workflow commerce2customer --target-keyspace customer reversetraffic --tablet-types "rdonly,replica"
E1009 20:59:35.000461   35915 main.go:56] rpc error: code = Unknown desc = no streams found in keyspace commerce for commerce2customer_reverse

❯ vtctlclient MoveTables -- reversetraffic customer.commerce2customer --tablet_types "rdonly,replica"
W1008 19:45:47.734875   67717 log.go:39] Failed to read in config : Config File "vtconfig" Not Found in "[/Users/matt/git/vitess/examples/local]". This is optional, and can be ignored if you are not using config files. For a detailed explanation, see https://github.com/vitessio/vitess/blob/main/doc/viper/viper.md#config-files.
reversetraffic was successful for workflow customer.commerce2customer
Start State: All Reads Switched. Writes Not Switched
Current State: Reads Not Switched. Writes Not Switched
```

I think that it makes sense to backport this *at least* to v21 so that this is addressed in a minimum of 1 GA branches/releases. For those on older releases, the workaround is to use the equivalent `vtctlclient` command. In the end, what this really did is closer align the two implementations around how this work is done.

### Manual test

```
alias vtctldclient='command vtctldclient --server=localhost:15999'
cd examples/local

./101_initial_cluster.sh; mysql < ../common/insert_commerce_data.sql; ./201_customer_tablets.sh; ./202_move_tables.sh; ./203_switch_reads.sh

echo -e "\nReversing traffic for reads..."
vtctldclient MoveTables --workflow commerce2customer --target-keyspace customer reversetraffic --tablet-types "rdonly,replica"
echo
```

With the client command output on the PR branch being:
```
SwitchTraffic was successful for workflow customer.commerce2customer

Start State: Reads Not Switched. Writes Not Switched
Current State: All Reads Switched. Writes Not Switched


Reversing traffic for reads...
ReverseTraffic was successful for workflow customer.commerce2customer

Start State: All Reads Switched. Writes Not Switched
Current State: Reads Not Switched. Writes Not Switched
```

## Related Issue(s)

  - Fixes: https://github.com/vitessio/vitess/issues/16918

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by mattlord** [comment]
@rohit-nayak-ps thanks again for the help with the new tests! That was a life saver. I thought about what else to test, now that we have the new and updated e2e and unit tests and nothing obvious came to mind. If you have general suggestions please let me know and I'll work on adding/updating tests for those conditions. In the meantime I'll think about what gaps might still be left myself (haven't come up with anything yet).

File: go/vt/vtctl/workflow/server.go [hunk 1]
```diff
@@ -3138,17 +3138,22 @@ func (s *Server) WorkflowSwitchTraffic(ctx context.Context, req *vtctldatapb.Wor
 	span, ctx := trace.NewSpan(ctx, "workflow.Server.WorkflowSwitchTraffic")
 	defer span.Finish()
 
+	if req == nil {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "invalid nil request")
+	}
+
```
Comment (deepthi) [inline go/vt/vtctl/workflow/server.go:3144]: Why do we need to add this? Is it even possible for `req` to be nil?
If we _do_ want to add a nil check, we should be doing it in all other similar functions in this file too - WorkflowDelete etc.

**Review by deepthi — APPROVED** [review]

File: go/vt/vtctl/workflow/server.go (hunk 1 above)
Comment (mattlord) [reply go/vt/vtctl/workflow/server.go:3144]: I could remove it, along with the unit test case covering it. We don't have to do it here as the `VtctldServer` grpc server does have a panic handler that would catch the case where it is nil and we try to dereference it by accessing the request's fields.

Comment (mattlord) [reply go/vt/vtctl/workflow/server.go:3144]: I could remove it, along with the unit test case covering it. We don't have to do it here as the VtctldServer grpc server does have a panic handler that would catch the case where it is nil and we try to dereference it by accessing the request's fields. https://github.com/vitessio/vitess/blob/main/go/vt/vtctl/grpcvtctldserver/server.go#L5290-L5306

Comment (mattlord) [reply go/vt/vtctl/workflow/server.go]: Done.

**Review by rohit-nayak-ps — APPROVED** [review]

**Pull request merged**